cleanrl/sac_continuous_action_jax.py

```diff
@@ -0,0 +1,548 @@
+
+    @nn.compact
+    def __call__(self, x: jnp.ndarray, action: jnp.ndarray) -> jnp.ndarray:
+        x = jnp.concatenate([x, action], -1)
+        x = nn.Dense(self.n_units)(x)
+        x = nn.relu(x)
+        x = nn.Dense(self.n_units)(x)
+        x = nn.relu(x)
+        x = nn.Dense(1)(x)
+        return x
+
+
+class VectorCritic(nn.Module):
+    n_units: int = 256
+    n_critics: int = 2
+
+    @nn.compact
+    def __call__(self, obs: jnp.ndarray, action: jnp.ndarray) -> jnp.ndarray:
+        # Idea taken from https://github.com/perrin-isir/xpag
+        # Similar to https://github.com/tinkoff-ai/CORL for PyTorch
+        vmap_critic = nn.vmap(
+            Critic,
+            variable_axes={"params": 0},  # parameters not shared between the critics
+            split_rngs={"params": True},  # different initializations
+            in_axes=None,
+            out_axes=0,
+            axis_size=self.n_critics,
+        )
+        q_values = vmap_critic(
+            n_units=self.n_units,
+        )(obs, action)
+        return q_values
+
+
+class Actor(nn.Module):
+    action_dim: Sequence[int]
+    n_units: int = 256
+    log_std_min: float = -20
+    log_std_max: float = 2
+
+    @nn.compact
+    def __call__(self, x: jnp.ndarray) -> tfd.Distribution:
+        x = nn.Dense(self.n_units)(x)
+        x = nn.relu(x)
+        x = nn.Dense(self.n_units)(x)
+        x = nn.relu(x)
+        mean = nn.Dense(self.action_dim)(x)
+        log_std = nn.Dense(self.action_dim)(x)
+        log_std = jnp.clip(log_std, self.log_std_min, self.log_std_max)
+        dist = TanhTransformedDistribution(
+            tfd.MultivariateNormalDiag(loc=mean, scale_diag=jnp.exp(log_std)),
+        )
+        return dist
+
+
+@partial(jax.jit, static_argnames="actor")
+def sample_action(
+    actor: Actor,
+    actor_state: TrainState,
+    observations: jnp.ndarray,
+    key: jax.random.KeyArray,
+) -> jnp.array:
+    dist = actor.apply(actor_state.params, observations)
+    action = dist.sample(seed=key)
+    return action
+
+
+@partial(jax.jit, static_argnames="actor")
+def select_action(actor: Actor, actor_state: TrainState, observations: jnp.ndarray) -> jnp.array:
+    return actor.apply(actor_state.params, observations).mode()
+
+
+def scale_action(action_space: gym.spaces.Box, action: np.ndarray) -> np.ndarray:
+    """
+    Rescale the action from [low, high] to [-1, 1]
+    (no need for symmetric action space)
+
+    :param action: Action to scale
+    :return: Scaled action
+    """
+    low, high = action_space.low, action_space.high
+    return 2.0 * ((action - low) / (high - low)) - 1.0
+
+
+def unscale_action(action_space: gym.spaces.Box, scaled_action: np.ndarray) -> np.ndarray:
+    """
+    Rescale the action from [-1, 1] to [low, high]
+    (no need for symmetric action space)
+
+    :param scaled_action: Action to un-scale
+    """
+    low, high = action_space.low, action_space.high
+    return low + (0.5 * (scaled_action + 1.0) * (high - low))
```

Comment:
The doc string can go to the documentation site, like done in TD3's documentation 
<img width="802" alt="image" src="https://user-images.githubusercontent.com/5555347/199145182-923ad71c-c8be-4c31-91c8-e2563ca136aa.png">

Reply:
I named them like because `scale_action` produces a `scaled_action` (so between [-1, 1]) and `unscale_action` produces an `unscaled_action` (between [low, high], and for that it needs the scale and bias to transform to the unormalized space).
I did that because it's easier to always work with scaled actions and unscale them at the very last moment (to prevent issues like https://github.com/vwxyzjn/cleanrl/issues/279).

I could rename them to `normalize_action`/`denormalize_action` (or `unnormalize`)?